sunpy/timeseries/sources/goes.py

```diff
@@ -214,14 +184,58 @@ def _parse_hdus(cls, hdulist):
                              ('xrsb', u.W/u.m**2)])
         return data, header, units
 
+    @staticmethod
+    def _parse_netcdf(filepath):
+        """
+        Parses the netCDF GOES files to return the data, header and associated units.
+
+        Parameters
+        ----------
+        filepath : `~str`
+            The path of the file to parse
+        """
+        d = h5netcdf.File(filepath, mode="r")
+
+        header =  MetaDict(OrderedDict(d.attrs))
+        if "a_flux" in d.variables:
+            xrsa = np.array(d["a_flux"])
+            xrsb = np.array(d["b_flux"])
+            start_time_str = d["time"].attrs["units"].astype(str).lstrip("seconds since").rstrip("UTC")
+            times = parse_time(start_time_str) + TimeDelta(d["time"], format="sec")
+        elif "xrsa_flux" in d.variables:
+            xrsa = np.array(d["xrsa_flux"])
+            xrsb = np.array(d["xrsb_flux"])
+            start_time_str = d["time"].attrs["units"].astype(str).lstrip("seconds since")
+            times = parse_time(start_time_str) + TimeDelta(d["time"], format="sec")
+
+        else:
+            raise ValueError(f"The file {filepath} doesn't seem to be a GOES netcdf file.")
+
+        d.close()
```

Comment:
If neither `a_flux` or `xrsa_flux` are in `d.variables` and the `ValueError` is raised, doesn't this mean that the file handle will be left open? I would still argue for a context manager here, but maybe I'm just being overly cautious. 

```suggestion
        with h5netcdf.File(filepath, mode="r") as d:
            header =  MetaDict(OrderedDict(d.attrs))
            if "a_flux" in d.variables:
                xrsa = np.array(d["a_flux"])
                xrsb = np.array(d["b_flux"])
                start_time_str = d["time"].attrs["units"].astype(str).lstrip("seconds since").rstrip("UTC")
                times = parse_time(start_time_str) + TimeDelta(d["time"], format="sec")
            elif "xrsa_flux" in d.variables:
                xrsa = np.array(d["xrsa_flux"])
                xrsb = np.array(d["xrsb_flux"])
                start_time_str = d["time"].attrs["units"].astype(str).lstrip("seconds since")
                times = parse_time(start_time_str) + TimeDelta(d["time"], format="sec")
            else:
                raise ValueError(f"The file {filepath} doesn't seem to be a GOES netcdf file.")
```